include/zephyr/posix/sys/stat.h

```diff
@@ -34,10 +34,29 @@
 extern "C" {
 #endif
 
+#if defined(__XT_CLANG__) && defined(CONFIG_NEWLIB_LIBC)
+/*
+ * Xtensa xt-clang is based on old newlib which does not _timespec.h.
+ * And some of the typedefs below have already been defined, but
+ * there are no corresponding *_DECLARED macros, so we need to define
+ * them manually to avoid them being re-declared.
+ */
+#define HAS__TIMESPEC_H 0
```

Comment:
Vague concerns about a name in the 'application' namespace, but prefixing it with '_' would risk conflicting with a C library define.